backend/src/db/pool.ts

```diff
@@ -1,11 +1,15 @@
 import { Pool } from 'pg';
 
-const pool = new Pool({
-    user: process.env.POSTGRES_USER,
-    host: 'localhost',
-    database: process.env.POSTGRES_DB,
-    password: process.env.POSTGRES_PASSWORD,
-    port: parseInt(process.env.POSTGRES_PORT ?? '5432'),
-});
+const databaseConfig = () => {
```

Comment:
Why is this a function that returns an object and not an object? Do we need multiple instances of the config object that can be modified without affecting the other instances?